.github/workflows/run-tx-tool.yaml

```diff
@@ -68,30 +59,36 @@ jobs:
           echo "ZCASH_NODE_PORT: $ZCASH_NODE_PORT"
           echo "ZCASH_NODE_PROTOCOL: $ZCASH_NODE_PROTOCOL"
           echo "IMAGE_TAG: $IMAGE_TAG"
+          # Define and persist the full ECR image path
+          echo "ECR_IMAGE=$ECR_PUBLIC_REGISTRY/$ECR_REGISTRY_ALIAS/$ECR_REPOSITORY:$IMAGE_TAG" >> $GITHUB_ENV
+          echo "ECR_IMAGE: $ECR_PUBLIC_REGISTRY/$ECR_REGISTRY_ALIAS/$ECR_REPOSITORY:$IMAGE_TAG"
 
       - name: Pull Docker image from ECR
-        env:
-          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
         run: |
-          docker pull $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
+          docker pull $ECR_IMAGE
 
       - name: Run Docker container
-        env:
-          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
         run: |
           docker run \
             -e ZCASH_NODE_ADDRESS="$ZCASH_NODE_ADDRESS" \
             -e ZCASH_NODE_PORT="$ZCASH_NODE_PORT" \
             -e ZCASH_NODE_PROTOCOL="$ZCASH_NODE_PROTOCOL" \
-            $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
+            $ECR_IMAGE
 
-      - name: Notify Slack
-        if: always() # Ensures this step runs regardless of success or failure of previous steps
+      - name: Notify Slack on Success
+        if: success() # Runs only if the previous steps succeed
         uses: rtCamp/action-slack-notify@v2
         env:
           SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
           SLACK_TITLE: "tx-tool"
-          SLACK_MESSAGE: "Finished running tx-tool"
-          SLACK_MESSAGE_ON_SUCCESS: "✅ Docker run completed successfully with Docker tag '${{ steps.set-defaults.outputs.image_tag }}'."
-          SLACK_MESSAGE_ON_FAILURE: "🚨 Docker run failed with Docker tag '${{ steps.set-defaults.outputs.image_tag }}'. Check logs in Github for details."
+          SLACK_MESSAGE: "✅ Docker run completed successfully with Docker tag '${{ steps.set-defaults.outputs.image_tag }}'."
+          SLACK_COLOR: ${{ job.status }}
+
+      - name: Notify Slack on Failure
+        if: failure() # Runs only if the previous steps fail
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_FAIL }}
```

Comment:
the are still `secrets` here, what will happen if they are not present?

Reply:
think that we need to check if they are present to avoid loking like the script fail. For a repo after clone.